### PR TITLE
Clean up FunctionDef class

### PR DESCRIPTION
**Pull request by mostafaMamoni**
how i cleaned up Functiondef class:
put sys.exit(property/method name) in every property/method. so every exit status of pyccel has the name of a property/method, and then every property/method still contains sys.exit(). It's definitely useless according to the current tests.

### CODE REVIEW

File: pyccel/ast/core.py [hunk 1]
```diff
@@ -3167,6 +3168,7 @@ def local_vars(self):
 
     @property
     def global_vars(self):
+        sys.exit("global vars in functiondef\n")
         return self._global_vars
```
Comment (saidctb) [inline pyccel/ast/core.py:3172]: we need to keep this one, in order to have access to the global variables used in the function

File: pyccel/ast/core.py [hunk 2]
```diff
@@ -3183,6 +3185,7 @@ def kind(self):
 
     @property
     def is_static(self):
+        sys.exit("is static in functiondef\n")
```
Comment (saidctb) [inline pyccel/ast/core.py:3188]: this can probably be removed

File: pyccel/ast/core.py [hunk 3]
```diff
@@ -3238,13 +3241,15 @@ def doc_string(self):
         return ""
 
     def print_body(self):
+        sys.exit("print body in functiondef\n")
```
Comment (saidctb) [inline pyccel/ast/core.py:3244]: this also

File: pyccel/ast/core.py [hunk 4]
```diff
@@ -3277,6 +3282,7 @@ def rename(self, newname):
         self._name = newname
 
     def vectorize(self, body , header):
```
Comment (saidctb) [inline pyccel/ast/core.py:3284]: we need to keep this in order to vectorize a function

**Review by saidctb — APPROVED** [review]

File: pyccel/ast/core.py (hunk 4 above)
Comment (EmilyBourne) [reply pyccel/ast/core.py]: @saidctb Is the vectorizing currently implemented? It doesn't look like this function does the vectorizing. Are we sure that this is how the vectorizing will be done when it is implemented?

File: pyccel/ast/core.py [hunk 5]
```diff
@@ -3374,12 +3362,10 @@ def __reduce_ex__(self, i):
 
     # TODO
     def check_pure(self):
-        sys.exit("check_pure in functiondef\n")
         raise NotImplementedError('')
 
     # TODO
     def check_elemental(self):
-        sys.exit("check_elemental in functiondef\n")
         raise NotImplementedError('')
```
Comment (EmilyBourne) [inline pyccel/ast/core.py]: @saidctb Presumably these functions can also be removed as they are not even implemented?

Comment (yguclu) [reply pyccel/ast/core.py]: I agree. These two functions must be removed!

File: pyccel/ast/core.py (hunk 4 above)
Comment (yguclu) [reply pyccel/ast/core.py]: I am also perplexed by this method, which is doing operations that do not make much sense to me.
For example, the line
```python
self._kind = procedure
```
should read
```python
self._kind = 'procedure'
```
Further, I don't understand why `body` should be an argument to this method, it looks like the vectorization is really performed by another function that we do not have. And finally, we do not even have a `vectorize` decorator!

@saidctb I am strongly in favor of removing this function... We can copy it to some "obsolete" folder if you really care about it.

Comment (saidctb) [reply pyccel/ast/core.py]: it's okay we can delete it

File: pyccel/ast/core.py (hunk 5 above)
Comment (saidctb) [reply pyccel/ast/core.py]: yes

**Review by EmilyBourne — APPROVED** [review]

**Comment by EmilyBourne** [comment]
@mostafaMamoni  Does this PR fix an issue (eg #550)? If so, please mark it as such either by modifying the PR description or by linking the issue manually using the bar on the right hand side of the page

**Review by yguclu — APPROVED** [review]

**Pull request merged**